File: fixtures/typescript/type_inference/overloads/overloads.ts
export function combine(a: number, b: number): number;
export function combine(a: string, b: string): string;
export function combine(a: number, b: string): string;
export function combine(a: number | string, b: number | string): number | string {
  if (typeof a === "number" && typeof b === "number") {
    return a + b;
  }
  return String(a) + String(b);
}

export function run(): string {
  const ints = combine(1, 2);
  const strs = combine("a", "b");
  const mixed = combine(1, "b");
  return `${ints}${strs}${mixed}`;
}

File: fixtures/kotlin/type_inference/overloads/Overloads.kt
package app

class Overloads {
    fun combine(a: Int, b: Int): Int = a + b

    fun combine(a: String, b: String): String = a + b

    fun combine(a: Int, b: String): String = "$a$b"

    fun run(): String {
        val ints = combine(1, 2)
        val strs = combine("a", "b")
        val mixed = combine(1, "b")
        return "$ints$strs$mixed"
    }
}
